packages/components/modules/profiles/ProfilePopover/AddProfileModal/index.tsx

```diff
@@ -0,0 +1,179 @@
+  addNewProfileLabel = 'New organization',
+  termsAndConditionsUrl = '',
+  addNewProfileDescription = 'Create an organization and invite multiple members to manage and collaborate.',
+  submitLabel = 'Create Organization',
+  onClose,
+  open,
+  setOpen,
+  userId,
+}) => {
+  const form = useForm<OrganizationCreateForm>({
+    defaultValues: {
+      name: '',
+      urlPath: '',
+    },
+    resolver: zodResolver(schema),
+  })
+
+  const [commitMutation, isMutationInFlight] = useOrganizationCreateMutation()
+
+  const handleClose = () => {
+    setOpen(false)
+    form.reset()
+    onClose?.()
+  }
+
+  const onSubmit = (data: OrganizationCreateForm) => {
+    if (isMutationInFlight) return
+
+    nextClientMutationId += 1
+    const clientMutationId = nextClientMutationId.toString()
+
+    const connections = ConnectionHandler.getConnectionID(userId, 'ProfilesListFragment_profiles')
+
+    commitMutation({
+      variables: {
+        input: {
+          name: data.name,
+          urlPath: data.urlPath,
+          clientMutationId,
+        },
+        connections: [connections],
+      },
+      onCompleted: (response, errors) => {
+        if (errors) {
+          console.error(errors)
+          return
+        }
+        const mutationErrors = response?.organizationCreate?.errors
+        setFormRelayErrors(form, mutationErrors)
+
+        if (!mutationErrors?.length) {
+          handleClose()
+        }
+      },
+      onError: console.error,
+    })
+  }
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleClose}
+      aria-labelledby="organization-modal-title"
+      aria-describedby="organization-modal-description"
+      PaperProps={{
+        sx: {
+          borderRadius: { xs: 0, sm: 2 },
+          minWidth: { xs: '100%', sm: 0 },
+          margin: { xs: 0, sm: 2 },
+          maxWidth: { sm: 366 },
+          height: { xs: '100%', sm: 'auto' },
+          minHeight: { xs: '100%', sm: 'auto' },
+        },
+      }}
+    >
+      <DialogTitle id="organization-modal-title">
+        <Box
+          sx={{ display: 'flex', alignItems: 'center', gap: 2, justifyContent: 'space-between' }}
+        >
+          <Typography variant="h6">{addNewProfileLabel}</Typography>
+          <IconButton onClick={handleClose}>
+            <CloseIcon />
+          </IconButton>
+        </Box>
+      </DialogTitle>
+      <Form onSubmit={form.handleSubmit(onSubmit)}>
+        <DialogContent
+          sx={{ display: 'flex', flexDirection: 'column', gap: 4 }}
+          id="organization-modal-description"
+        >
+          <Typography color="text.secondary">{addNewProfileDescription}</Typography>
+          <Box display="flex" flexDirection="column" gap={2}>
+            <TextField
+              label="Name"
+              name="name"
+              size="medium"
+              control={form.control}
+              onChange={() => {
+                form.trigger('name')
```

Comment:
just curious why we have this trigger, if we set `useForm({mode: "onChange"})` wouldn't that be enough?